howtos/module/_posts/2016-06-30-Ten-ways-improve-modules.md

```diff
@@ -0,0 +1,110 @@
+Hopefully, these tips will increase your development speed and save you lots and lots of time. And if not satisfied, you'll get your money back! (Oh, wait...)  
+
+## The time savers
+Here is a list of very very useful tools and features (if you ask me). Prepare for some technical stuff as we're going to cover some of them in depth.
+
+### 1. Use an Integrated Development Environment
+Using an integrated development environment such as PhpStorm or NetBeans is a great way to improve your productivity.
+Tools such as these come with dozens of nifty features out-of-the-box to develop quicker... a lot quicker!
+
+As the name implies, they intend to integrate as many features as possible, allowing you for example to automatically upload your changes over FTP. It is also possible make a direct connection with the MySQL database and browse it from your IDE (not recommended for production environments!) to see what was actually written to the database by your module.  
+An IDE is aware of your code, guiding you while you write your code, wherever and whenever you need it. Cool tip for PhpStorm users: navigate to *Help > Productivity guide*. It will show you how many keystrokes it has saved you, as well as amount of bugs prevented and hours saved. The numbers will add up quickly after using it for a while!
+These IDEs were designed to never take your hands off the keyboard while performing basic programming tasks. Feel free to configure the keyboard shortcuts in a way you're comfortable with and don't disable the "Tip of the day" messages. They can make the "steep" learning curve of using an IDE a little bit less steep.
+
+### 2. Stick to coding standards
+PrestaShop has changed their coding standards several times in the past and as of this month, another coding standard has gone into effect. Why does PrestaShop change its [coding standards]({% post_url 2016-05-10-prestashop-coding-standards %})?  
+It is mostly because PrestaShop is an open source projects and needs several standards to abide to, because otherwise the whole codebase would become an unreadable mess and the GitHub repository would become a place full of (small) conflicts --  no time to actually focus on the software itself.
+("What color is [the bikeshed](http://bikeshed.com/)?", "I think it should be [PrestaShop purple](http://251B5B.bikeshed.com/)!" -- just to name a few examples!)  
+  
+Even though these standards are very specific to open source, you can incorporate them into your own workflow as well, in order to increase productivity.
+There are probably numerous times where a co-worker might have to look at your work and need to adjust your code. Having someone else take a look at your code may apply even if you're a one man army coder.
+Using standards makes it much easier to cooperate. Don't forget that these standards were also designed to make the code somewhat "predictably" and allows for less mistakes, thus making the code you write less error-prone. Isn't that cool?  
```

Comment:
Are you saying that it would have been...

![6fr8c2vt](https://cloud.githubusercontent.com/assets/2833749/16523337/36d99e68-3fa3-11e6-9662-57c6d6fc9c4a.jpg)

...predictable?

![yeaaaaaaaaaaaaaaaaaaaaaaaaaaaa-55c9eacf8d9f9](https://cloud.githubusercontent.com/assets/2833749/16523361/516d4608-3fa3-11e6-9d95-1a8cd6a9e36e.png)